tests/preprocessing/test_data_preprocessor.py

```diff
@@ -21,170 +31,160 @@ def sample_data():
     }
     return pd.DataFrame(data)
 
-
 @pytest.fixture
-def data_processor(project_config, mocker):
-    mocker.patch(
-        "power_consumption.preprocessing.data_preprocessor.DataProcessor.load_data"
-    )
-    processor = DataProcessor(config=project_config)
-    return processor
-
+def data_processor(project_config, sample_data):
+    """
+    Create a DataProcessor instance with sample data for testing.
+
+    Parameters
+    ----------
+    project_config : Config
+        The project configuration.
+    sample_data : pd.DataFrame
+        The sample data for testing.
+
+    Returns
+    -------
+    DataProcessor
+        An instance of DataProcessor initialized with sample data.
+    """
+    return DataProcessor(config=project_config, data=sample_data)
 
 @pytest.fixture
-def data_processor_with_data(data_processor, sample_data):
-    data_processor.data = sample_data
+def mock_spark_session(mocker):
```

Comment:
oh wow, even mock spark session! very cool!

Reply:
Normally I do not like mocking if I want to test specific functionality. I usually create a spark or delta_spark fixture in my conftest file to actually have and use a spark session for tests:
```python
import pytest
from delta import configure_spark_with_delta_pip
from pyspark.sql import SparkSession


@pytest.fixture(scope="module")
def spark():
    spark_builder = SparkSession.builder.appName("Test")
    spark_builder.config("spark.master", "local[1]")
    spark = spark_builder.getOrCreate()
    yield spark
    spark.stop()


@pytest.fixture()
def delta_spark_session():
    spark_builder = SparkSession.builder.appName("DeltaTableTest")
    spark_builder.config(
        "spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension"
    )
    spark_builder.config(
        "spark.sql.catalog.spark_catalog",
        "org.apache.spark.sql.delta.catalog.DeltaCatalog",
    )
    spark_builder.config("spark.master", "local[1]")
    spark = configure_spark_with_delta_pip(spark_builder).getOrCreate()
    yield spark
    spark.stop()

```

But in this case, it was expecting a Databricks Spark Session, and I just gave up and mocked it instead 😆